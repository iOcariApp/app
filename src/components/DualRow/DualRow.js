import React from "react";
import { View } from "react-native";
import PropTypes from "prop-types";
import styles from "./DualRow.style";

const DualRow = ({ left, right, separation, style }) => (
  <View style={[styles.main, style]}>
    <View style={[styles.column, { marginRight: separation }]}>{left}</View>
    <View style={styles.column}>{right}</View>
  </View>
);

DualRow.propTypes = {
  left: PropTypes.node,
  right: PropTypes.node,
  separation: PropTypes.number,
  style: PropTypes.oneOfType([PropTypes.object, PropTypes.number]),
};
DualRow.defaultProps = {
  separation: 0,
  style: {},
};

export default DualRow;
