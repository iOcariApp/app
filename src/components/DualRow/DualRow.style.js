import { StyleSheet } from "react-native";

export default StyleSheet.create({
  main: {
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "flex-start",
    width: "100%",
  },
  column: {
    flex: 1,
  },
});
