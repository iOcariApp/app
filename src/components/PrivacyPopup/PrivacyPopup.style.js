import { StyleSheet } from "react-native";

const textColor = "#4E4E4E";

export default StyleSheet.create({
  dialogStyle: {
    width: 288,
    height: 396,
    paddingTop: 29,
    paddingBottom: 35,
    paddingLeft: 31,
    paddingRight: 31,
  },
  main: {
    flex: 1,
    flexDirection: "column",
    justifyContent: "space-between",
    alignItems: "center",
  },
  title: {
    fontSize: 20,
    color: textColor,
    fontWeight: "500",
    lineHeight: 24,
    textAlign: "center",
  },
  text: {
    fontSize: 14,
    color: textColor,
    lineHeight: 18,
    textAlign: "center",
  },
});
