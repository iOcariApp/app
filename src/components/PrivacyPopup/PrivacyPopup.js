import React from "react";
import PropTypes from "prop-types";
import { View, Text, Image } from "react-native";
import styles from "./PrivacyPopup.style";

import privacyShield from "assets/privacy-shield/privacy-shield.png";

import PopupDialog, { SlideAnimation } from "react-native-popup-dialog";

import DualRow from "components/DualRow";
import PinkEmptyButton from "components/Button/PinkEmptyButton";
import PinkButton from "components/Button/PinkButton";

const fromTopAnimation = new SlideAnimation({
  slideFrom: "top",
});

class PrivacyPopup extends React.PureComponent {
  render = () => {
    const { refCallback, readMore, accept } = this.props;
    return (
      <PopupDialog
        ref={refCallback}
        dialogAnimation={fromTopAnimation}
        dismissOnTouchOutside={false}
        width={288}
        height={396}
        dialogStyle={{
          paddingTop: 29,
          paddingBottom: 35,
          paddingLeft: 31,
          paddingRight: 31,
        }}
      >
        <View
          style={{
            flex: 1,
            flexDirection: "column",
            justifyContent: "space-between",
            alignItems: "center",
          }}
        >
          <Image source={privacyShield} />
          <Text
            style={{
              fontSize: 20,
              color: "#4E4E4E",
              fontWeight: "500",
              lineHeight: 24,
              textAlign: "center",
            }}
          >
            Tu seguridad es importante
          </Text>
          <Text
            style={{
              fontSize: 14,
              color: "#4E4E4E",
              lineHeight: 18,
              textAlign: "center",
            }}
          >
            En nuestra política de privacidad de datos y términos y condiciones
            te explicamos claramente como manejamos tu información y como te
            ayudamos a tener la mejor experiencia de juego, mirálo antes de
            proceder.
          </Text>
          <DualRow
            left={<PinkEmptyButton small title="Revisar" onPress={readMore} />}
            right={<PinkButton small title="Acepto" onPress={accept} />}
            separation={5}
          />
        </View>
      </PopupDialog>
    );
  };
}

PrivacyPopup.propTypes = {
  refCallback: PropTypes.func.isRequired,
  readMore: PropTypes.func.isRequired,
  accept: PropTypes.func.isRequired,
};

export default PrivacyPopup;
