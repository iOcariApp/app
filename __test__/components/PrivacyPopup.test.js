import React from "react";
import renderer from "react-test-renderer";

import PrivacyPopup from "components/PrivacyPopup";
import DualRow from "components/DualRow";

const props = {
  refCallback: jest.fn(),
  readMore: jest.fn(),
  accept: jest.fn(),
};

const wrapper = shallow(<PrivacyPopup {...props} />);
const buttons = wrapper.find(DualRow).dive();

test("renders correctly", () => {
  const tree = renderer.create(<PrivacyPopup {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("`readMore` props gets called when review button is pressed", () => {
  const button = buttons.childAt(0).childAt(0);

  button.simulate("press");

  expect(props.readMore).toHaveBeenCalledTimes(1);
});

test("`accept` props gets called when accept button is pressed", () => {
  const button = buttons.childAt(1).childAt(0);

  button.simulate("press");

  expect(props.accept).toHaveBeenCalledTimes(1);
});
