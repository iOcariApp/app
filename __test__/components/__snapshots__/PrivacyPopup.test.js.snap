// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "zIndex": 1000,
      },
      Object {
        "height": 0,
        "left": 0,
        "top": -10000,
        "width": 0,
      },
      Object {
        "height": 1334,
        "width": 750,
      },
      null,
    ]
  }
>
  <View
    collapsable={undefined}
    pointerEvents="none"
    style={
      Object {
        "backgroundColor": "#000",
        "flex": 1,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "top": 0,
        "width": 750,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "height": 1334,
          "left": 0,
          "opacity": 1,
          "position": "absolute",
          "top": 0,
          "width": 750,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    />
  </View>
  <View
    collapsable={undefined}
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderRadius": 8,
        "height": 396,
        "paddingBottom": 35,
        "paddingLeft": 31,
        "paddingRight": 31,
        "paddingTop": 29,
        "transform": Array [
          Object {
            "translateY": -800,
          },
        ],
        "width": 288,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "space-between",
        }
      }
    >
      <Image
        source={1}
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#4E4E4E",
            "fontSize": 20,
            "fontWeight": "500",
            "lineHeight": 24,
            "textAlign": "center",
          }
        }
      >
        Tu seguridad es importante
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#4E4E4E",
            "fontSize": 14,
            "lineHeight": 18,
            "textAlign": "center",
          }
        }
      >
        En nuestra política de privacidad de datos y términos y condiciones te explicamos claramente como manejamos tu información y como te ayudamos a tener la mejor experiencia de juego, mirálo antes de proceder.
      </Text>
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            },
            Object {},
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {
                "marginRight": 5,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "marginLeft": 15,
                  "marginRight": 15,
                },
                undefined,
                Array [
                  Object {
                    "marginBottom": 0,
                    "marginLeft": 0,
                    "marginRight": 0,
                    "marginTop": 0,
                    "width": "100%",
                  },
                  Object {},
                ],
                Object {
                  "borderRadius": 4,
                },
              ]
            }
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={null}
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View
                pointerEvents="box-only"
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#9E9E9E",
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "padding": 19,
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "borderRadius": 4,
                    },
                    Object {
                      "padding": 12,
                    },
                    undefined,
                    undefined,
                    undefined,
                    Array [
                      Object {
                        "height": 40,
                      },
                      Object {
                        "borderColor": "#EF5865",
                        "borderWidth": 2,
                      },
                    ],
                    undefined,
                    undefined,
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {},
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      Array [
                        Object {
                          "color": "white",
                          "fontSize": 20,
                        },
                        Object {
                          "color": "#EF5865",
                        },
                        Object {
                          "fontSize": 17.5,
                        },
                        Object {
                          "fontSize": 18,
                        },
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                      ],
                    ]
                  }
                >
                  Revisar
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "marginLeft": 15,
                  "marginRight": 15,
                },
                undefined,
                Array [
                  Object {
                    "marginBottom": 0,
                    "marginLeft": 0,
                    "marginRight": 0,
                    "marginTop": 0,
                    "width": "100%",
                  },
                  Object {},
                ],
                Object {
                  "borderRadius": 4,
                },
              ]
            }
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={null}
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View
                pointerEvents="box-only"
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#9E9E9E",
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "padding": 19,
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    Object {
                      "backgroundColor": "#EF5865",
                    },
                    Object {
                      "borderRadius": 4,
                    },
                    Object {
                      "padding": 12,
                    },
                    undefined,
                    undefined,
                    undefined,
                    Array [
                      Object {
                        "height": 40,
                      },
                      Object {},
                    ],
                    undefined,
                    undefined,
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {},
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      Array [
                        Object {
                          "color": "white",
                          "fontSize": 20,
                        },
                        undefined,
                        Object {
                          "fontSize": 17.5,
                        },
                        Object {
                          "fontSize": 18,
                        },
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                      ],
                    ]
                  }
                >
                  Acepto
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
